tidy3d/components/data/monitor_data.py

```diff
@@ -1069,46 +1068,33 @@ def to_adjoint_field_sources(self, fwidth: float) -> List[CustomCurrentSource]:
 
         sources = []
 
-        # Define source geometry based on coordinates in the data
-        data_mins = []
-        data_maxs = []
+        source_geo = self.monitor.geometry
+        freqs = self.monitor.freqs
 
-        def shift_value(coords) -> float:
-            """How much to shift the geometry by along a dimension (only if > 1D)."""
-            return SHIFT_VALUE_ADJ_FLD_SRC if len(coords) > 1 else 0
-
-        for _, field_component in self.field_components.items():
-            coords = field_component.coords
-            data_mins.append({key: min(val) + shift_value(val) for key, val in coords.items()})
-            data_maxs.append({key: max(val) + shift_value(val) for key, val in coords.items()})
-
-        rmin = []
-        rmax = []
-        for dim in "xyz":
-            rmin.append(max(val[dim] for val in data_mins))
-            rmax.append(min(val[dim] for val in data_maxs))
-
-        source_geo = Box.from_bounds(rmin=rmin, rmax=rmax)
-
-        # Define source dataset
-        # Offset coordinates by source center since local coords are assumed in CustomCurrentSource
-
-        for freq0 in tuple(self.field_components.values())[0].coords["f"]:
+        for freq0 in freqs:
             src_field_components = {}
             for name, field_component in self.field_components.items():
+                # get the VJP values at frequency and apply adjoint phase
                 field_component = field_component.sel(f=freq0)
-                forward_amps = field_component.values
-                values = -1j * forward_amps
+                values = -1j * field_component.values
+
+                # make source go backwards
+                if "H" in name:
+                    values *= -1
```

Comment:
Is this what we do in the adjoint plugin too?

I am trying to think about whether this is the correct thing to do in all cases. It's probably possible to see it somehow from the adjoint formulation if written for the H field. I think this is correct. I'm just wondering a little because the intuition of this minus sign "making source go backwards" makes sense for a real propagating mode, but it gets a bit murkier for evanescent or lossy modes. The other option is to take np.conj(values). If the E field is real and the mode is lossless and propagating, then the H field is imaginary and the two operations are the same. In general the `np.conj(values)` operation would be like time reversal while the `values *= -1` operation is like reflection of the mode in the source plane, which produce different results in some cases.

I do think the `-1` is correct here though but do you remember if it comes out of the adjoint math?